libs/wire-api/src/Wire/API/OAuth.hs

```diff
@@ -0,0 +1,597 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2022 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Wire.API.OAuth where
+
+import Cassandra hiding (Set)
+import Control.Lens (preview, view, (.~), (?~), (^.))
+import Control.Monad.Except
+import Crypto.JWT hiding (Context, params, uri, verify)
+import qualified Data.Aeson.KeyMap as M
+import qualified Data.Aeson.Types as A
+import Data.ByteString.Conversion
+import Data.ByteString.Lazy (toStrict)
+import qualified Data.HashMap.Strict as HM
+import Data.Id as Id
+import Data.Range
+import Data.Schema
+import qualified Data.Set as Set
+import Data.String.Conversions (cs)
+import Data.Swagger (ToParamSchema (..))
+import qualified Data.Swagger as S
+import qualified Data.Text as T
+import Data.Text.Ascii
+import qualified Data.Text.Encoding as TE
+import Data.Text.Encoding.Error as TErr
+import Data.Time
+import Imports hiding (exp, head)
+import Servant hiding (Handler, JSON, Tagged, addHeader, respond)
+import Servant.Swagger.Internal.Orphans ()
+import Test.QuickCheck (Arbitrary)
+import URI.ByteString
+import Web.FormUrlEncoded (Form (..), FromForm (..), ToForm (..), parseUnique)
+import Wire.API.Error
+import Wire.Arbitrary (GenericUniform (..))
+import qualified SAML2.WebSSO as configs
```

Comment:
what's this?  doesn't compile for me locally.

Reply:
yes, don't know where this came from, but it's gone now: [5b07b11](https://github.com/wireapp/wire-server/pull/2989/commits/5b07b111c96e92fda50f94739dfddad08a7e4a32)